### PR TITLE
ADD: Post question system tests

### PR DESCRIPTION
**Pull request by VladimirMikulic**
I've refactored post_question_test.rb file to log in the user before each test because the user can't submit a question if he is not logged in.

Tests:
 - posting a question (https://github.com/publiclab/PublicLab.Editor/issues/113)
 - form validation messages

### CODE REVIEW

**Comment by VladimirMikulic** [comment]
I've fixed @Uzay-G's test that would mess up other system tests due to improper elements selecting.

File: test/system/post_test.rb [hunk 1]
```diff
@@ -124,27 +124,30 @@ def assert_page_does_not_reload(message = "page should not reload")
...
-    old_wiki_content = find("#content").text
+    old_wiki_content = find("#content p").text
 
     find("a#edit-btn").click()
     find("#text-input").set("wiki text")
     find("a#publish").click()
 
     # view wiki
-    current_wiki_content = find("#content").text
+    current_wiki_content = find("#content p").text
```
Comment (Uzay-G) [inline test/system/post_test.rb]: Sorry this might pass in testing but it doesn't accomplish the required purpose. The `#content` div is composed of different elements like headings, paragraphs and many others. Not just p tags

Comment (VladimirMikulic) [reply test/system/post_test.rb]: Not really, there is only one p tag, I can add :first-of-type if you insist though.

Comment (VladimirMikulic) [reply test/system/post_test.rb]: Still better than `#content` alone.

Comment (Uzay-G) [reply test/system/post_test.rb]: For example, if you go to https://publiclab.org/about/. You will see the #content div has many different elements. Is there a way to get the text of all those elements and not just the p tags?

Comment (VladimirMikulic) [reply test/system/post_test.rb]: This is testing page.

Comment (Uzay-G) [reply test/system/post_test.rb]: Alright; but say later someone edits the test and adds a heading. The test will fail because the p tags will not have changed so the content will be marked equal

Comment (VladimirMikulic) [reply test/system/post_test.rb]: No, it won't fail, the `p` tag is first and it would fetch it.

Comment (VladimirMikulic) [reply test/system/post_test.rb]: It failed again :(

Comment (VladimirMikulic) [reply test/system/post_test.rb]: I think it's because of `visit`, let's try again by setting fixed path.

File: test/system/post_test.rb [hunk 2]
```diff
@@ -119,32 +119,33 @@ def assert_page_does_not_reload(message = "page should not reload")
     # Make sure that image has been uploaded
     page.assert_selector('#preview img', count: 1)
   end
-  
+
   test "changing and reverting versions works correctly for wiki" do
-    wiki = nodes(:wiki_page)
+    visit '/wiki/wiki-page-path/'
```
Comment (Uzay-G) [inline test/system/post_test.rb]: Why did you change this? Also could you please explain how my test is affecting others?

Comment (VladimirMikulic) [reply test/system/post_test.rb]: I am trying to figure that out. Every time I submit a PR with system tests, your test fails.
Let's wait for Travis, but it might be possible that it's fetching the wrong row, so I'll have to update it again.

Comment (VladimirMikulic) [reply test/system/post_test.rb]: On localhost, it works fine though.

**Comment by VladimirMikulic** [comment]
Finally, I've fixed your bug @Uzay-G.
You were asserting the text with `find` method dynamically fetching the element and of course it fails.
Also, please do fix #7303, my tests (and I believe tests from others) too were failing because of this issue.

Update your repo and be careful next time you are writing tests.

**Comment by VladimirMikulic** [comment]
#7303 needs to be patched ASAP. Travis failed again due to it. @SidharthBansal ?

**Comment by VladimirMikulic** [comment]
Ready to be merged :tada: :rocket:
